.mailmap

```diff
@@ -1,292 +1,387 @@
-# Prevent git from showing duplicate names with commands like "git shortlog"
-# See the manpage of git-shortlog for details.
-# The syntax is:
-# Name that should be used <email that should be used> Bad name <bad email>
+# Before editing this file, run bin/authors_update.py. This will
+# make sure that all authors known to git are added to AUTHORS
+# and inform you about the types of entries needed here.
 #
-# You can skip Bad name if it is the same as the one that should be used, and is unique.
+# Entries in this file are made for two reasons:
 #
-# This file is up-to-date if the command git log --format="%aN <%aE>" | sort -u
-# gives no duplicates.
-Ondřej Čertík <ondrej@certik.cz> Ondrej Certik <ondrej@certik.cz>
+# 1) to make the name and address of an author appear canonically
+# 2) to merge authors in git that are actually the same person
+#
+# To address either of these issues, add an entry to this file like
+#
+#     Canonical Name <Canonical@email> commit name <commit@email>
+#
+# then
+#    1) iteratively re-run bin/authors_update.py and make edits
+#       here until no error is raised by authors_update
+#    2) run bin/mailmap_update.py to check the results
+#    3) as a final check, commit only changes to .mailmap and
+#       re-run bin/authors_update.py to see all the
+#       authors that will be added, e.g.
+#         $ git add .mailmap; git commit -m mailmap
+#         $ git reset --hard; bin/authors_update.py
+#    4) when all names and addresses in AUTHORS appear as
+#       desired and each entry respresents a distint author
+#       and there is no error when running bin/mailmap_update.py
+#       then commit the results together with the .mailmap changes:
+#         $ git reset head~n (if you've made n commits of changes)
+#         $ git commit -am 'Update .mailmap/AUTHORS. Welcome to SymPy!'
+#
+# NOTES:
+#
+# While human arbitration must always select the canonical names and
```

Comment:
You could expand on "human arbitration" here (meaning, if you are unsure, ask the person, and that everyone has the right to decide how their name is spelled and which email is canonically used here).

Reply:
Is this any better?
```
# Although the authors_update.py script incorporates some warnings
# and error messages, there is no fully automated way to make entries
# in this file since the goal is to assocate one or more git names
# and addresses with a given author according to the author's
# preference (and that preference may differ from anything in git).
# The following notes may be useful in guiding one in making entries
# entries in this file.
```

Reply:
That's fine. So far our social policies on this have been somewhat lax. I readily admit that when I update this file, I usually don't contact people unless something really isn't clear. I instead use other signals like what info is on the person's GitHub account and which metadata was used most recently. Should we contact everyone to get their preferences? Probably, but it's a lot of work to do, and takes a while for people to get back. Should we be more conservative about adding someone's full name from their GitHub account to .mailmap when the git metadata only has a username?  

My take is that at the end of the day, it's quite easy to fix, so it isn't a big deal. But if others feel that a more formal policy here is important we can add something.

Reply:
> Should we contact everyone to get their preferences?

I should perhaps phrase this in another way. I am just trying to say that a human needs to look at the options and (lacking author input) pick the one that makes the most sense. It would be impossible to get this right 100% with an automated script.